otri/analysis/find_clusters.py

```diff
@@ -0,0 +1,97 @@
+__version__ = "0.1"
+__author__ = "Riccardo De Zen <riccardodezen98@gmail.com>"
+
+from otri.utils import key_handler as kh
+from otri.filtering.stream import Stream
+
+from typing import Set, Sequence, Any, Callable
+import time
+from . import Analysis
+
+from otri.filtering.filter_net import EXEC_AND_PASS, FilterNet
+from otri.filtering.filter_layer import FilterLayer
+from otri.filtering.filters.generic_filter import GenericFilter
+from otri.filtering.filters.nuplicator_filter import NUplicatorFilter
+
+from otri.validation.validators.cluster_validator import ClusterValidator
+
+
+class ClusterAnalysis(Analysis):
+
+    def __init__(self, keys: Set[str], cluster_size: int, on_output: Callable):
+        '''
+        Parameters:
+            keys : Set[str]
+                The keys that may contain clusters.
+
+            on_output : Callable
+                A function requiring no parameters to call every time the network outputs something.
+        '''
+        self.keys = keys
+        self.cluster_size = cluster_size
+        self.on_output = on_output
+
+    def execute(self, in_streams: Sequence[Stream]) -> Any:
+        '''
+        Starts data analyis.\n
+
+        Parameters:
+            in_streams : Stream
+                Required Streams: single Stream from the database.
+
+        Returns:
+            Tuple of values.
+            result: The whole state of the network.
+            flagged: How many atoms were flagged.
+            total: How many atoms got through.
+            elapsed_time: How much the computation took, in seconds.
+        '''
+        stream_per_key = [key for key in self.keys]
+        output_per_key = [key + "_out" for key in self.keys]
+
+        start_time = time.time()
+        analysis_net = FilterNet([
+            FilterLayer([
+                # Tuple extractor
+                GenericFilter(
+                    inputs="db_tuples",
+                    outputs="db_atoms",
+                    operation=lambda element: element[0]
+                )
+            ], EXEC_AND_PASS),
+            FilterLayer([
+                # To Lowercase
+                GenericFilter(
+                    inputs="db_atoms",
+                    outputs="lower_atoms",
+                    operation=lambda atom: kh.lower_all_keys_deep(atom)
```

Comment:
Codacy found an issue: [Lambda may not be necessary](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=6277039)

Reply:
Listen to the wise codacy.